crypto/x509/x509_compat_test.cc

```diff
@@ -0,0 +1,297 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <gtest/gtest.h>
+
+#include <openssl/x509.h>
+
+#include "../internal.h"
+#include "../test/test_util.h"
+#include "../test/x509_util.h"
+
+/*
+
+The default root certificate key, "ROOT_KEY_1", used for self-signed
+roots, unless otherwise specified. This is EC prime256v1 key.
+-----BEGIN PRIVATE KEY-----
+MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQgfVMH4tqIaJ6OzyxY
+mqWXNwmK7gpXYDFhX80mXKgzrGGhRANCAATCqXrfbdTjFimzdBHxj71Ejcc/stea
+5xAU/xxK+s77yXzB5lfy/zEbcYxuOrnwHrWsX9sugWgCy74ZRNWJPTDW
+-----END PRIVATE KEY-----
+
+The default end-entity certificate key, "EE_KEY_1", used for ee certs,
+unless otherwise specified. This is EC prime256v1 key.
+-----BEGIN PRIVATE KEY-----
+MIGHAgEAMBMGByqGSM49AgEGCCqGSM49AwEHBG0wawIBAQQgfVMH4tqIaJ6OzyxY
+mqWXNwmK7gpXYDFhX80mXKgzrGGhRANCAATCqXrfbdTjFimzdBHxj71Ejcc/stea
+5xAU/xxK+s77yXzB5lfy/zEbcYxuOrnwHrWsX9sugWgCy74ZRNWJPTDW
+-----END PRIVATE KEY-----
+*/
+
+/*
+This self-signed root certificate's basicConstraints extension
+cA:false. The root certificate though has the keyCertSign bit set for the
+keyUsage extension. This is in violation of RFC 5280 4.2.1.9:
+"If the cA boolean is not asserted, then the keyCertSign bit in the key
+usage extension MUST NOT be asserted"
```

Comment:
NP: Is there anything unique about the parsing of the certs being tested here? Should they also be added to the `pkcs7_verify` fuzz corpus?

Reply:
Parsing no, but I do want to make a pass separately to flesh out our fuzzer with some of these newer certs as seeds.